### PR TITLE
Multiple bug fixes, refactoring, CartToDto mapper

### PR DESCRIPTION
**Pull request by epadams**
- Checks for negative stock (#29)
- Stock is set, not added to (#32)
- CreateCart returns DTO now (#30)
- Creating new item will respect initial stock value (#31)
- Refactoring in controller to make use of new `CartDto.ToDTO()` function

### CODE REVIEW

File: ShopAPI/ShopAPI/Services/ProductService.cs [hunk 1]
```diff
@@ -119,11 +119,15 @@ public async Task<Product> UpdateProductPriceAsync(int id, decimal price)
         /// <returns>Nothing.</returns>
         public async Task<Product> AddProductStockAsync(int id, int quantity)
         {
+            if (quantity < 0 || quantity == null)
+            {
+                throw new ConflictException("Quantity must be greater than or equal to 0");
```
Comment (jbytes1027) [inline ShopAPI/ShopAPI/Services/ProductService.cs]: Should be [422 Unprocessable Content](https://www.rfc-editor.org/rfc/rfc9110.html#name-422-unprocessable-content)

File: ShopAPI/ShopAPI/Services/ProductService.cs [hunk 2]
```diff
@@ -119,11 +119,15 @@ public async Task<Product> UpdateProductPriceAsync(int id, decimal price)
         /// <returns>Nothing.</returns>
         public async Task<Product> AddProductStockAsync(int id, int quantity)
         {
+            if (quantity < 0 || quantity == null)
```
Comment (jbytes1027) [inline ShopAPI/ShopAPI/Services/ProductService.cs]: Will never be null, remove check.

File: ShopAPI/ShopAPI/Services/ProductService.cs [hunk 3]
```diff
@@ -119,11 +119,15 @@ public async Task<Product> UpdateProductPriceAsync(int id, decimal price)
         /// <returns>Nothing.</returns>
         public async Task<Product> AddProductStockAsync(int id, int quantity)
```
Comment (jbytes1027) [inline ShopAPI/ShopAPI/Services/ProductService.cs]: Rename to `UpdateProductStockAsync`

**Review by jbytes1027 — CHANGES_REQUESTED** [review]
Fix AddProductStockAsync

File: ShopAPI/ShopAPI/Services/ProductService.cs (hunk 1 above)
Comment (epadams) [reply ShopAPI/ShopAPI/Services/ProductService.cs]: Done

File: ShopAPI/ShopAPI/Services/ProductService.cs (hunk 2 above)
Comment (epadams) [reply ShopAPI/ShopAPI/Services/ProductService.cs]: Done

File: ShopAPI/ShopAPI/Services/ProductService.cs (hunk 3 above)
Comment (epadams) [reply ShopAPI/ShopAPI/Services/ProductService.cs]: Done

**Review by jbytes1027 — APPROVED** [review]

**Pull request merged**